draft-ietf-taps-interface.md

```diff
@@ -944,38 +944,55 @@ capacity that it sees fit.
-is sent by passing a Message Object and additional parameters
-{{send-params}} to the Send Action on an established Connection:
+Once a Connection has been established, it can be used for sending data. Data is
+sent in terms of Messages, which allow the application to communicate the boundaries
+of the data being transferred. By default, Send enqueues a complete Message,
+and takes optional per-Message parameters (see {{send-basic}}). All Send actions
+are asynchronous, and deliver events (see {{send-events}}). Sending partial 
+Messages for streaming large data is also supported (see {{send-partial}}).
+
+## Basic Sending {#send-basic}
+
+The most basic form of sending on a connection involves enqueuing a single Data
+block as a complete Message, with default send parameters. Message data is
+created as an array of octets, and the resulting object contains both the byte
+array and the length of the array.
 
 ~~~
-Connection.Send(Message, sendParameters)
+messageData := "hello".octets()
+Connection.Send(messageData)
 ~~~
 
-The type of the Message to be passed is dependent on the implementation, and
-on the constraints on the Protocol Stacks implied by the Connection's
-transport parameters. It may itself contain an array of octets to be
-transmitted in the transport protocol payload, or be transformable to an array
-of octets by a sender-side framer (see {{send-framing}}).
+The interpretation of a Message to be sent is dependent on the implementation, and 
+on the constraints on the Protocol Stacks implied by the Connection’s transport parameters.
+For example, a Message may be a single datagram for UDP Connections; or an HTTP
+Request for HTTP Connections.
 
 Some transport protocols can deliver arbitrarily sized Messages, but other
 protocols constrain the maximum Message size. Applications can query the
-protocol property Maximum Message Size on Send to determine the maximum size.
-
-There may also be system and Protocol Stack dependent limits on the size of
-a Message which can be transmitted atomically. For that reason, the Message
-object passed to the Send action may also be a partial Message, either
-representing the whole data object and information about the range of bytes
-to send from it, or an object referring back to the larger whole Message.
-The details of partial Message sending are implementation-dependent.
+protocol property Maximum Message Size on Send to determine the maximum size
+allowed for a single Message. If a Message is too large to fit in the Maximum Message
+Size for the Connection, the Send will fail with a SendError event ({{send-error}}). For
+example, it is invalid to send a Message over a UDP connection that is larger than
+the available datagram sending size.
 
 If Send is called on a Connection which has not yet been established, an
 Initiate Action will be implicitly performed simultaneously with the Send.
 Used together with the Idempotent property (see {{send-idempotent}}), this can
```

Comment:
A nit: I suggest to remove the "Used" at the beginning of this sentence

Reply:
Agreed, removed.